curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-intermediate-algorithmic-thinking-by-building-a-dice-game/64b45a195557d801660f07d5.md

```diff
@@ -0,0 +1,410 @@
+---
+id: 64b45a195557d801660f07d5
+title: Step 57
+challengeType: 0
+dashedName: step-57
+---
+
+# --description--
+
+You should write an `if` statement to check against the `selectedValue` variable. Inside the `if` statement reassign `rolls` to `0`, increment `round` by `1`.Then, you will call the functions named `updateStats` and `resetRadioOption`. Finally, you will call the function named `updateScore` and pass in arguments `selectedValue` and `achieved`.
+
+# --hints--
+
+You should create an `if` statement to check against the `selectedValue` variable.
+
+```js
+assert.match(code, /if\s*\(\s*selectedValue\s*\)\s*{\s*[^}]*\s*}/);
+```
+
+Inside the `if` statement, you will reassign `rolls` to `0`, increment `round` by `1`.
+
+```js
+assert.match(code, /.*if\s*\(\s*selectedValue\s*\)\s*{\s*rolls\s*=\s*0\s*;?\s*round\s*\+\+\s*;?.*}/s);
+```
+
+You will call the functions named `updateStats` and `resetRadioOption`.
+
+```js
+assert.match(code, /if\s*\(\s*selectedValue\s*\)\s*{\s*rolls\s*=\s*0\s*;?\s*round\s*\+\+\s*;?\s*updateStats\s*\(\s*\)\s*;?\s*resetRadioOption\s*\(\s*\).*}/s);
+```
+
+You will call the function named `updateScore` and pass in arguments `selectedValue` and `achieved`.
```

Comment:
There's no `updateScore` function defined at this point. I think either definition should be moved before this step, or adding the `updateScore` call here should be added after function is defined.